st-exerciseviewer/src/main/kotlin/de/saring/exerciseviewer/data/ExerciseCadence.kt

```diff
@@ -5,10 +5,12 @@ package de.saring.exerciseviewer.data
  *
  * @property cadenceAvg Average cadence of exercise (rpm).
  * @property cadenceMax Maximum cadence of exercise (rpm).
+ * @property cadenceTotal Total cycles of exercise recorded.
  *
  * @author Stefan Saring
  */
 data class ExerciseCadence(
 
     var cadenceAvg: Short,
-    var cadenceMax: Short)
+    var cadenceMax: Short,
+    var cyclesTotal: Short)
```

Comment:
IMHO the cyclesTotal attribute is optional, it's only available for users of specific Garmin devices. That's why the type Short? should be used. It's better to display nothing than 0, the information 0 is not valid.

You can pass the default value "null" to the constructor, because most parser implementations can't provide the number of total cycles. So the cyclesTotal-handling needs to be added just to the Garmin FIT parser.

Reply:
Tried to implement that, however ran into a problem.
`var cyclesTotal: Short?`
does not compile with the error

> OptionalPanelController.kt: (79, 83) Only safe (?.) or non-null asserted (!!.) calls are allowed on a nullable receiver of type Short?

Sorry, Kotlin noob. Can you advice what's going wrong

Alex